.github/workflows/go.yml

```diff
@@ -60,6 +60,8 @@ jobs:
     needs: download
     runs-on: ubuntu-latest
     name: Update coverage badge
+    permissions:
+      contents: write
```

Comment:
What is the rationale to add content write permissions in the Github workflow?

Reply:
This adds write permission to the default GitHub token `github.token`.

Reply:
...which may be the default. Being explicit is better than implicit.

Reply:
Example
![image](https://github.com/Boeing/config-file-validator/assets/952007/542a660d-4d76-46b9-ac8c-7fc56a05d887)

Current value
https://github.com/Boeing/config-file-validator/settings/actions

Reply:
Here is the solution. Repository-wide write permissin 🤯 
https://github.com/ad-m/github-push-action/issues/189#issuecomment-1816532638

Reply:
Thanks for the analysis @szepeviktor - that's really helpful

Reply:
Second one is also done.
I use `git diff` in many repositories. Could you tell me how it misbehaves?

Reply:
I believe the issue is that `git diff --quiet` exits 1 when there is a changed file whereas `git status --porcelain` exits 0. For example here is the output of both commands after I made a manual change to `index.md`:

```bash
>> vim index.md 
>> git diff --quiet
>> echo $?
1
>> git status --porcelain
 M index.md
>> echo $?
0
```

Reply:
I see!
Actually `git status` does not changes its exit status on file change.
Please consider using `git diff --quiet`.